cloud/scope/powervs_cluster_test.go

```diff
@@ -3830,3 +3832,799 @@ func TestCreateVPCSubnet(t *testing.T) {
+			},
+			resourceType:   infrav1beta2.ResourceTypeTransitGateway,
+			expectedResult: false,
+		},
+		{
+			name: "When resourceType is TransitGateway and TransitGateway status is not nil",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+					Status: infrav1beta2.IBMPowerVSClusterStatus{
+						TransitGateway: &infrav1beta2.TransitGatewayStatus{
+							ControllerCreated: ptr.To(true),
+						},
+					},
+				},
+			},
+			resourceType:   infrav1beta2.ResourceTypeTransitGateway,
+			expectedResult: true,
+		},
+		{
+			name: "When resourceType is DHCPServer and DHCPServer status is nil",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			},
+			resourceType:   infrav1beta2.ResourceTypeDHCPServer,
+			expectedResult: false,
+		},
+		{
+			name: "When resourceType is DHCPServer and DHCPServer status is not nil",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+					Status: infrav1beta2.IBMPowerVSClusterStatus{
+						DHCPServer: &infrav1beta2.ResourceReference{
+							ControllerCreated: ptr.To(true),
+						},
+					},
+				},
+			},
+			resourceType:   infrav1beta2.ResourceTypeDHCPServer,
+			expectedResult: true,
+		},
+		{
+			name: "When resourceType is COSInstance and COSInstance status is nil",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			},
+			resourceType:   infrav1beta2.ResourceTypeCOSInstance,
+			expectedResult: false,
+		},
+		{
+			name: "When resourceType is COSInstance and COSInstance status is not nil",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+					Status: infrav1beta2.IBMPowerVSClusterStatus{
+						COSInstance: &infrav1beta2.ResourceReference{
+							ControllerCreated: ptr.To(true),
+						},
+					},
+				},
+			},
+			resourceType:   infrav1beta2.ResourceTypeCOSInstance,
+			expectedResult: true,
+		},
+		{
+			name: "When resourceType is not valid",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			},
+			resourceType:   infrav1beta2.ResourceTypePublicGateway,
+			expectedResult: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		g := NewWithT(t)
+		t.Run(tc.name, func(_ *testing.T) {
+			isResourceCreated := tc.clusterScope.isResourceCreatedByController(tc.resourceType)
+			g.Expect(isResourceCreated).To(Equal(tc.expectedResult))
+		})
+	}
+}
+
+func TestDeleteCOSInstance(t *testing.T) {
+	var (
+		mockResourceController *mockRC.MockResourceController
+		mockCtrl               *gomock.Controller
+	)
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockResourceController = mockRC.NewMockResourceController(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When cos instance resource is not created by controller", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("When COS instance resource is not created by controller", func(t *testing.T) {
```
Better to keep the names uniform throughout

Reply:
Done